MonoGame.Framework/Graphics/Texture2D.DirectX.cs

```diff
@@ -86,80 +88,92 @@ private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat
+                        // Some drivers may add pitch to rows.
+                        // We need to copy each row separatly and skip trailing zeros.
+                        stream.Seek(0, SeekOrigin.Begin);
 
-                        var elementSize = _format.GetSize();
-                        if (_format.IsCompressedFormat())
+                        var elementSizeInByte = ReflectionHelpers.SizeOf<T>.Get();
+                        for (var row = 0; row < rows; row++)
                         {
-                            // for 4x4 block compression formats an element is one block, so elementsInRow
-                            // and number of rows are 1/4 of number of pixels in width and height of the rectangle
-                            elementsInRow /= 4;
-                            rows /= 4;
-                        }
-                        var rowSize = elementSize * elementsInRow;
-                        if (rowSize == databox.RowPitch)
-                            stream.ReadRange(data, startIndex, elementCount);
-                        else
-                        {
-                            // Some drivers may add pitch to rows.
-                            // We need to copy each row separatly and skip trailing zeros.
-                            stream.Seek(0, SeekOrigin.Begin);
-
-                            var elementSizeInByte = ReflectionHelpers.SizeOf<T>.Get();
-                            for (var row = 0; row < rows; row++)
-                            {
-                                int i;
-                                for (i = row * rowSize / elementSizeInByte; i < (row + 1) * rowSize / elementSizeInByte; i++)
-                                    data[i + startIndex] = stream.Read<T>();
-
-                                if (i >= elementCount)
-                                    break;
-
-                                stream.Seek(databox.RowPitch - rowSize, SeekOrigin.Current);
-                            }
+                            int i;
+                            for (i = row * rowSize / elementSizeInByte; i < (row + 1) * rowSize / elementSizeInByte; i++)
+                                data[i + startIndex] = stream.Read<T>();
+
+                            if (i >= elementCount)
+                                break;
+
+                            stream.Seek(databox.RowPitch - rowSize, SeekOrigin.Current);
                         }
                     }
-                    finally
-                    {
-                        SharpDX.Utilities.Dispose( ref stream);
-                    }
+                }
+                finally
+                {
+                    SharpDX.Utilities.Dispose( ref stream);
+
+                    d3dContext.UnmapSubresource(_cachedStagingTexture, 0);
                 }
             }
         }
 
+        protected override void Dispose(bool disposing)
+        {
+            SharpDX.Utilities.Dispose(ref _cachedStagingTexture);
```

Comment:
I think you need to put the `if (disposing)` here as well to implement the disposing pattern correctly.

@KonajuGames ?